spec/controllers/users_controller_spec.rb

```diff
@@ -68,4 +69,24 @@
       end
     end
   end
+
+  describe "#destroy" do
+    let(:deletable) { FactoryGirl.create :alice }
+    before(:each) do
+      sign_in admin
+    end
+    describe "destroying users" do
+      it "should not delete user with GenericFiles" do
+        @generic_file = GenericFile.create { |gf| 
+          gf.apply_depositor_metadata(user) 
+        }
+        expect( number_of_deposits(user) ).to eq(1)
+        expect { user.destroy }.not_to change(User, :count)  
```

Comment:
You're calling #destroy on the user *model*, but you created a method called ```destroy``` in the User _Controller_. You've got the right instinct in this test -- that is, that this is model behaviour that you want to enforce everywhere, not just in one controller -- but it's a model test, not a controller test, so it's in the wrong directory.

Fundamentally, I don't think you want that ```destroy``` code in the controller at all. What you want to do is create a ```before_destroy``` callback in the user *model* that returns false if-and-only-if the user has deposits. [The Rails Guide on the ActiveRecord Callback Lifecycle](http://guides.rubyonrails.org/active_record_callbacks.html) has a lot of good details (ActiveFedora copies the ActiveRecord Callbacks more-or-less verbatim).